presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/PushAggregationThroughOuterJoin.java

```diff
@@ -292,7 +315,21 @@ private MappedAggregationInfo createAggregationOverNull(AggregationNode referenc
                 AggregationNode.Step.SINGLE,
                 Optional.empty(),
                 Optional.empty());
-        return new MappedAggregationInfo(aggregationOverNullRow, aggregationsSymbolMapping);
+
+        return Optional.of(new MappedAggregationInfo(aggregationOverNullRow,
+                                aggregationsSymbolMapping));
+    }
+
+    private static boolean checkSymbolDependency(AggregationNode.Aggregation aggregation, Map<Symbol, SymbolReference> sourcesSymbolMapping)
+    {
+        FunctionCall functionCall = aggregation.getCall();
```

Comment:
This could be inlined to:

```
return sourcesSymbolMapping.keySet().stream().anyMatch(currentArgument ->
                aggregation.getCall().getArguments().contains(currentArgument.toSymbolReference()));
```

But most readable would be:

```
List<Symbol> functionArguments =  aggregation.getCall().getArguments();
return sourcesSymbolMapping.keySet().stream()
     .map(Symbol::toSymbolReference)
     .anyMatch(functionArguments::contains);
```